### ISSUE TITLE
Can you add higher order derivative support for torch's embedding function? 

### ISSUE THREAD

**Issue by lixilinx** [pytorch/pytorch#50226]
## 🚀 Feature
add higher order derivative support to torch.nn.functional.embedding

## Motivation

Almost all native Pytorch operations support higher order derivative. However, I checked torch.nn.functional.embedding and found that it does not. It will be a nice feature to have.

My torch version is 1.8. Please run the following code to confirm my observation.
```
import torch

embedding_matrix = torch.rand(10, 3)
embedding_matrix.requires_grad_(True)
inputs = torch.tensor([[1,2,3], [4,5,6]])
y = torch.nn.functional.embedding(inputs, embedding_matrix)
g = torch.autograd.grad(torch.sum(y), embedding_matrix, create_graph=True)[0]
print(g.grad_fn) # is None
```

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer

**Comment by zou3519** [comment]
`embedding` does support higher order gradients:
```py
import torch
embedding_matrix = torch.rand(10, 3)
embedding_matrix.requires_grad_(True)
inputs = torch.tensor([[1,2,3], [4,5,6]])
y = torch.nn.functional.embedding(inputs, embedding_matrix)
g = torch.autograd.grad(torch.sum(y ** 2), embedding_matrix, create_graph=True)[0]
assert g.grad_fn is not None
```

The phenomenon that is happening in your example is that the values of `g` are independent from whatever values are in `embedding_matrix`; g is a constant function of `embedding_matrix`. In this case, it looks like autograd just decides not to assign a grad_fn:
```py
import torch
x = torch.randn(3, requires_grad=True)
# g(x) = 1., no matter what x is
g = torch.autograd.grad(torch.sum(x), x, create_graph=True)[0]
assert g.grad_fn is None
```

I don't know if this is the intended autograd behavior, but I think it has behaved like this for a while. @albanD can you confirm?

**Comment by albanD** [comment]
Hi,

Yes this is expected.
Independent gradients in pytorch can be represented by
- No graph (and so no grad_fn), leading to error if you try to backprop
- A graph that will generate 0 gradients
